### ISSUE TITLE
Unexpected restart causes Atom to lose project state

### ISSUE THREAD

**Issue by 50Wliu** [atom/atom#13757]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

I'm currently grappling with a problem with my laptop where it restarts when resuming from sleep without an external power source.  This causes Atom to lose the project I was working on, and I have to reopen it.  The Reopen Projects menu is also empty, though it is populated once a folder is opened.

### Steps to Reproduce

1. Open a folder in Atom
2. Somehow trigger an unexpected restart (I don't know if holding down the power button will work for this)
3. After restarting, reopen Atom

**Expected behavior:** The project to be displayed

**Actual behavior:** An empty editor with only an Untitled tab (if that option is enabled)

**Reproduces how often:** 100%

### Versions

```
Atom    : 1.15.0-dev-4eb3f22
Electron: 1.4.15
Chrome  : 53.0.2785.143
Node    : 6.5.0
```
```
apm  1.15.3
npm  3.10.5
node 4.4.5 x64
python 2.7.12
git 2.10.0.windows.1
visual studio 2015
```
Windows 10 x64.

### Additional Information

This may be related to #11406, though I have not been able to specifically reproduce that as 1) all open files are always saved before I put the laptop to sleep, and 2) there is no way for me to be typing in Atom when the restart occurs, as it happens as soon as Windows attempts to recover from sleep.

**Comment by ungb** [comment]
> The Reopen Projects menu is also empty, though it is populated once a folder is opened.

@50Wliu Do you know if there's another issue for this?  I'm seeing this as well.

**Comment by 50Wliu** [comment]
I don't think so.  I don't recall any Reopen Project-specific bugs being filed since it was added.
